test/unit/spec/EnrollSms_spec.js

```diff
@@ -262,59 +298,68 @@ function (Okta, OktaAuth, LoginUtil, Util, AuthContainer, Form, Beacon, Expect,
             });
           });
       });
+
       itp('clears previous errors in form when resend code', function () {
-        return setupAndSendInvalidCode().then(function (test) {
-          expect(test.form.hasErrors()).toBe(true);
-          expect(test.form.errorMessage()).toBe('Invalid Phone Number.');
-          return sendCodeOnEnter(test, successRes, 'US', '4151111111');
-        })
+        return setupAndSendInvalidCode()
+          .then(function (test) {
+            expect(test.form.hasErrors()).toBe(true);
+            expect(test.form.errorMessage()).toBe('Invalid Phone Number.');
+            sendCodeOnEnter(test, successRes, 'US', '4151111111');
+            return waitForEnrollActivateSuccess(test);
+          })
           .then(function (test) {
             expect(test.form.hasErrors()).toBe(false);
           });
       });
+
       itp('validation error if phone number field is blank', function () {
         return sendCodeFn(successRes, 'US', '')
           .then(function (test) {
             expect(test.form.hasErrors()).toBe(true);
           });
       });
 
-      itp('shows warning message to click "Re-send" after 30s', function () {  
-        Util.speedUpDelay();      
-        return setup().then(function (test) {
-          test.setNextResponse(resFactorEnrollActivateSms);
-          enterCode(test, 'US', '4151111111');
-          test.form.sendCodeButton().click();
-          return tick(test);
-        })
+      itp('shows warning message to click "Re-send" after 30s', function () {
+        Util.speedUpDelay();
+        return sendCodeFn(successRes, 'US', '4151111111')
+          .then(waitForEnrollActivateSuccess)
           .then(function (test) {
             expectResendButton(test);
             expect(test.form.hasWarningMessage()).toBe(true);
             expect(test.form.warningMessage()).toContain(
               'Haven\'t received an SMS? To try again, click Re-send code.');
-            return tick(test);
+            return test;
           })
           .then(function (test) {
             // Re-send will clear the warning
             $.ajax.calls.reset();
-            test.setNextResponse(resFactorEnrollActivateSms);
+            test.setNextResponse(successRes);
             test.form.sendCodeButton().click();
             expectSentButton(test);
             expect(test.form.hasWarningMessage()).toBe(false);
-            return tick(test);
+            return test;
           })
+          .then(waitForEnrollActivateSuccess)
           .then(function (test){
             // Re-send after 30s wil show the warning again
             expectResendButton(test);
             expect(test.form.warningMessage()).toContain(
               'Haven\'t received an SMS? To try again, click Re-send code.');
-          });     
+            jasmine.clock().uninstall();
```

Comment:
could cause interference with other tests?

Reply:
shall not but let me remove it since I didn't use clock and forgot to detele this line.